web/components/editor/mention-suggestion.ts

```diff
@@ -0,0 +1,61 @@
+import type { MentionOptions } from '@tiptap/extension-mention'
+import { ReactRenderer } from '@tiptap/react'
+import { User } from 'common/lib/user'
+import { searchInAny } from 'common/util/parse'
+import tippy from 'tippy.js'
+import { MentionList } from './mention-list'
+
+type Suggestion = MentionOptions['suggestion']
+
+// copied from https://tiptap.dev/api/nodes/mention#usage
+export const mentionSuggestion = (users: User[]): Suggestion => ({
+  items: ({ query }) =>
+    users.filter((u) => searchInAny(query, u.username, u.name)).slice(0, 5),
```

Comment:
Have you tried this on prod? My user filtering was fine on dev but super laggy on group adds bc I assume there were too many users to filter. Maybe it was something else though!